projects/common/src/utilities/angular/dynamic-component.service.ts

```diff
@@ -29,4 +33,32 @@ export class DynamicComponentService {
 
     return domPortalOutlet.attach(componentPortal);
   }
+
+  /**
+   * Dynamically create a component and insert it into the view container. Data and services can be provided to the dynamic
+   * component via `providers`
+   *
+   * @returns ComponentRef or undefined if the component could not be created
+   */
+  public attachComponentToViewContainer<T>(opts: DynamicComponentOptions<T>): ComponentRef<T> | undefined {
+    if (!isNil(opts.viewContainerRef) && !isNil(opts.componentClass)) {
```

Comment:
Why would we allow either to be nil? Can we enforce this by types instead so we can avoid returning undefined? Looks like componentClass already is, so that check is redundant, but not sure why viewContainerRef would be optional.

Reply:
Just bringing down the null check down to the lowest level so the consumer doesn't have to check.

Do you think its a good idea to let the consumer to first check and then call this method?

Reply:
The consumer has to check one or the other - if we fix the type, the compiler will force the consumer to check the input, but trust the output. If we leave the type and output optional, then the compiler will allow the consumer to trust the input, but check the output. My preference is for the former, it leaves the consumer less coupled with the impl details of the method - it doesn't need to care about what's required and what's not to get it to work, the method API tells is that.